### PR TITLE
ARROW-16912: [R][CI] Fix nightly centos package without GCS

### PR DESCRIPTION
**Pull request by nealrichardson**
cc @assignUser 

Most of the diff seems to be my editor trimming whitespace. The actual changes:

* Rename `r-nightly-packages` to `r-binary-packages` since they can be run on demand (not only nightly)
* Add it to the `r` crossbow group
* Turn ARROW_GCS=OFF in the centos-7 package. Where this setting happens is not obvious.

### CODE REVIEW

**Comment by nealrichardson** [comment]
@github-actions crossbow submit r-binary-packages

**Review by assignUser — APPROVED** [review]
👍 Thanks for the fix. 

👎 In regards to adding `r-binary-packages` to the r-group. I think we should take a closer look at the PR CI `r.yml` and see if we want to change it to be more in-line with `r-binary-packages` as they are quite similar instead of adding the task to the r-group. The heavy lifting happens in both jobs already (windows-cpp) and I think `r.yml` should have a macos build anyway.

File: docker-compose.yml
Comment (assignUser) [inline docker-compose.yml:413]: On second thought the logic should probably just be removed and the vars set under `environment:`

**Comment by nealrichardson** [comment]
> 👎 In regards to adding `r-binary-packages` to the r-group. I think we should take a closer look at the PR CI `r.yml` and see if we want to change it to be more in-line with `r-binary-packages` as they are quite similar instead of adding the task to the r-group. The heavy lifting happens in both jobs already (windows-cpp) and I think `r.yml` should have a macos build anyway.

I don't understand the objection. When I do a PR that affects different build settings in R, I want to run `crossbow submit -g r` and know that I've tested everything. To me, it doesn't make sense for that not to be exhaustive, even if it is somewhat wasteful. Yes, there is some redundancy, but this workflow does different work--clearly, otherwise I would have caught this in ARROW-16510, where I checked (what I thought was) all of the R workflows before merging.

File: docker-compose.yml
Comment (nealrichardson) [reply docker-compose.yml:413]: IIUC the intent is that these are only set on centos 7, and you can't conditionally set env vars in `environment:`. Though the setup appears to have baked in centos 7 from where it inherits, so maybe that's not a concern?

Comment (assignUser) [reply docker-compose.yml:413]: Yeah this service is fixed to centos 7 so we should be fine using `environment:`

**Comment by assignUser** [comment]
@nealrichardson Maybe I phrased this to strongly. It's not really an objection and should not block this PR. My intention was to start a discussion about/review of the `r.yml` CI and if it does what we want&need or if we should extend it.
You are right, it does make sense that all R related jobs are in the R group to prevent issues like this happening.

**Comment by nealrichardson** [comment]
> @nealrichardson Maybe I phrased this to strongly. It's not really an objection and should not block this PR. My intention was to start a discussion about/review of the `r.yml` CI and if it does what we want&need or if we should extend it. You are right, it does make sense that all R related jobs are in the R group to prevent issues like this happening.

👍 cool, we can review it. We've selected builds in the r.yml in order to catch the widest spectrum of issues with the fewest builds, but it's always worth evaluating if that's still the case. Currently, that's (1) separate C++ shared library build with everything on, force_tests: true; (2) bundled static build on centos-7/gcc 4.8; (3) windows, all rtools versions (we once pushed the rtools35 build to nightly but it kept failing because people would merge things that passed CI but broke the build, so we moved it back).

**Comment by assignUser** [comment]
With the current dearth of CI resources (atm 16 jobs queued 1 running) my thought of extending the `r.yml` to basically be `r-binary-packages` isn't really feasible.
Would it make sense to add one macOS build as that is not covered at all? Or are mac specific failures to rare for that to make sense?

**Comment by nealrichardson** [comment]
> Would it make sense to add one macOS build as that is not covered at all? Or are mac specific failures to rare for that to make sense?

macOS specific failures haven't been a big source of surprises on CI (as in, something gets merged that passes CI but breaks a nightly build). My guess why is (1) many of us working on the R bindings develop on macs so catch them locally and (2) if there's a problem in the C++ library, there are macOS C++ jobs that catch them. Whereas the centos-7 and rtools35 jobs (in particular) test things in the C++ library that aren't covered elsewhere.

**Comment by nealrichardson** [comment]
@github-actions crossbow submit r-binary-packages

**Pull request merged**
